catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DurableWebDavFileConsumer.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.util.List;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.component.file.GenericFileEndpoint;
+import org.apache.camel.component.file.GenericFileOperations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.sardine.Sardine;
+import com.github.sardine.SardineFactory;
+
+public class DurableWebDavFileConsumer extends AbstractDurableFileConsumer {
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(DurableFileSystemFileConsumer.class);
+
+    private DavAlterationObserver observer;
+
+    private EntryAlterationListener listener = new EntryAlterationListenerImpl();
+
+    private Sardine sardine = SardineFactory.begin();
+
+    DurableWebDavFileConsumer(GenericFileEndpoint<EventfulFileWrapper> endpoint,
+            Processor processor, GenericFileOperations<EventfulFileWrapper> operations) {
+        super(endpoint, processor, operations);
+    }
+
+    @Override
+    protected boolean pollDirectory(String fileName, List list, int depth) {
+        String remaining = ((DurableFileComponent) endpoint.getComponent()).remaining;
```

Comment:
❗️ I would recommend doing some more null checking in here. A DefaultEndpoint can be constructed with a null `component` and a DurableFileComponent can be constructed with a null `remaining`. I may be missing Javadoc that says those won't be null. Seems like `remaining` can only get populated if `DurableFileComponent.buildFileEndpoint()` is called.